src/display.cpp

```diff
@@ -83,28 +61,41 @@ void setup_display()
     // Show initial display buffer contents on the screen --
     // the library initializes this with an Adafruit splash screen.
     display.display();
-    delay(2000); // Pause for 2 seconds
+    delay(500); // Pause for short time
 
     // Clear the buffer
-    //display.clearDisplay();
+    display.clearDisplay();
 
     // Show the display buffer on the screen. You MUST call display() after
     // drawing commands to make them visible on screen!
     display.display();
 
-    int8_t f;
-
-    // Initialize 'snowflake' positions
-    for (f = 0; f < NUMFLAKES; f++)
-    {
-        icons[f][XPOS] = random(1 - LOGO_WIDTH, display.width());
-        icons[f][YPOS] = -LOGO_HEIGHT;
-        icons[f][DELTAY] = random(1, 6);
-        Serial.print(F("x: "));
-        Serial.print(icons[f][XPOS], DEC);
-        Serial.print(F(" y: "));
-        Serial.print(icons[f][YPOS], DEC);
-        Serial.print(F(" dy: "));
-        Serial.println(icons[f][DELTAY], DEC);
-    }
+    // Initialize lvgl library
+    lv_init();
+    lv_disp_draw_buf_init(&draw_buf, buf, NULL, SCREEN_WIDTH * 16);
+
+    // Initialize the display driver for lvgl
+    static lv_disp_drv_t disp_drv;
+    lv_disp_drv_init(&disp_drv);
+    disp_drv.hor_res = SCREEN_WIDTH;
+    disp_drv.ver_res = SCREEN_HEIGHT;
+    disp_drv.flush_cb = flushSSD1306Adafruit;
+    disp_drv.draw_buf = &draw_buf;
+
+    // Register display driver to lvgl
+    lv_disp_drv_register(&disp_drv);
+    delay(200);
+
+    // create first text in lvgl
+    lv_obj_t *label = lv_label_create(lv_scr_act());
+    lv_label_set_text(label, reinterpret_cast<const char *>(F("LVGL is up")));
```

Comment:
I am genuinely curious, why not simply:

```suggestion
    lv_label_set_text(label, "LVGL is up");
```

What is using `F()` good for? It looks AVR specific (using `<pgmspace.h>`) to me.